lib/libzfs/libzfs_mount.c

```diff
@@ -476,6 +384,8 @@ zfs_mount_at(zfs_handle_t *zhp, const char *options, int flags,
 	/*
 	 * Overlay mounts are disabled by default but may be enabled
 	 * via the 'overlay' property or the 'zfs mount -O' option.
+	 * This only applies to Linux.  Other platforms always allow overlay
+	 * mounts, so this flag is not required.
```

Comment:
I believe there's some confusion here.  The Linux VFS, like FreeBSDs, will always allow an overlay mount.  This code was originally added to emulate what I understand to be the illumos behavior of failing mounts when the mount point was not empty.  This was done in order to make ZFS to behave consistently across platforms (at least consistent with illumos, I didn't realize FreeBSD behaved like Linux).  From the man page for the "overlay" property.

```
     overlay=off|on
       Allow mounting on a busy directory or a directory which already con‐
       tains files or directories. This is the default mount behavior for
       Linux file systems.  For consistency with OpenZFS on other platforms
       overlay mounts are off by default. Set to on to enable overlay mounts.
```

Perhaps we could instead change the default value of the "overlay" property on FreeBSD and Linux so the behavior is consistent with other local filesystems.

Reply:
> Perhaps we could instead change the default value of the "overlay" property

I'd be happy with that. From a testing perspective, the tests should already have been adapted to check for this behavior when `is_freebsd`, so it should be a mostly simple matter of swapping those blocks and making the exception `is_illumos` or whatever platforms want the old default.